qt/widgets/common/test/FitDomainTest.h

```diff
@@ -0,0 +1,544 @@
+
+    TS_ASSERT_EQUALS(m_fitDomain->getFunction()->asString(),
+                     m_expDecay->asString());
+  }
+
+  void
+  test_that_removeFunction_will_not_throw_if_the_stored_function_is_a_nullptr() {
+    TS_ASSERT_THROWS_NOTHING(
+        m_fitDomain->removeFunction(m_flatBackground->asString()));
+  }
+
+  void
+  test_that_removeFunction_will_not_remove_a_function_if_the_function_specified_does_not_exist() {
+    m_fitDomain->setFunction(m_flatBackground);
+
+    TS_ASSERT_THROWS_NOTHING(
+        m_fitDomain->removeFunction(m_expDecay->asString()));
+    TS_ASSERT_EQUALS(m_fitDomain->getFunction()->asString(),
+                     m_flatBackground->asString());
+  }
+
+  void
+  test_that_addFunction_will_add_a_function_correctly_for_a_single_ifunction() {
+    m_fitDomain->addFunction(m_flatBackground);
+    TS_ASSERT_EQUALS(m_fitDomain->getFunction()->asString(),
+                     m_flatBackground->asString());
+  }
+
+  void
+  test_that_addFunction_will_add_a_second_function_correctly_to_create_a_composite() {
+    m_fitDomain->addFunction(m_flatBackground);
+    m_fitDomain->addFunction(m_expDecay);
+
+    TS_ASSERT_EQUALS(m_fitDomain->getFunction()->asString(),
+                     m_composite->asString());
+  }
+
+  void
+  test_that_addFunction_will_throw_if_attempting_to_create_a_nested_composite_function() {
+    m_fitDomain->addFunction(m_flatBackground);
+    TS_ASSERT_THROWS(m_fitDomain->addFunction(m_composite),
+                     std::invalid_argument const &);
+  }
+
+  void test_that_getParameterValue_will_get_the_parameter_value_if_it_exists() {
+    m_fitDomain->setFunction(m_flatBackground);
+    TS_ASSERT_EQUALS(m_fitDomain->getParameterValue("A0"), 0.0);
+  }
+
+  void
+  test_that_getParameterValue_will_throw_if_the_stored_function_is_a_nullptr() {
+    TS_ASSERT_THROWS(m_fitDomain->getParameterValue("A0"),
+                     std::runtime_error const &);
+  }
+
+  void
+  test_that_getParameterValue_will_throw_if_the_parameter_does_not_exist() {
+    m_fitDomain->setFunction(m_flatBackground);
+    TS_ASSERT_THROWS(m_fitDomain->getParameterValue("Height"),
+                     std::runtime_error const &);
+  }
+
+  void
+  test_that_setParameterValue_will_not_throw_if_the_stored_function_is_a_nullptr() {
+    TS_ASSERT_THROWS_NOTHING(m_fitDomain->setParameterValue("A0", 2.0));
+  }
+
+  void
+  test_that_setParameterValue_will_not_throw_if_the_stored_function_does_not_have_the_specified_parameter() {
+    m_fitDomain->setFunction(m_flatBackground);
+    TS_ASSERT_THROWS_NOTHING(m_fitDomain->setParameterValue("Height", 2.0));
+  }
+
+  void
+  test_that_setParameterValue_will_not_set_the_parameters_value_if_the_new_value_is_outside_the_constraints() {
+    m_fitDomain->setFunction(m_flatBackground);
+    m_fitDomain->updateParameterConstraint("", "A0", "0<A0<2");
+
+    m_fitDomain->setParameterValue("A0", 3.0);
+
+    TS_ASSERT_EQUALS(m_fitDomain->getParameterValue("A0"), 0.0);
```

Comment:
an answer of 0 is outside of the constraints as they it is greater than zero

Reply:
You are right. I think the notation used for constraints in mantid is a bit confusing. When it says `0<A0<2` it means it will set the lower bound to 0 and upper bound to 2, and it automatically assumes its inclusive!

This is probably something to change in the future to avoid confusion, but right now I see the rabbit hole being too deep. It shouldn't cause any problems.

Reply:
I think its just a problem with notation in constraint strings. The existing FunctionBrowser/Contraint stuff needs to be changed to use `<=` instead of `<`. Feels like a maintenance issue - I'll create one

Reply:
#30664

Reply:
thank you. It might be worth listing the bits of code you know will need changing.